### PR TITLE
Add support for anchor link items in `MasterMenu`

### PR DESCRIPTION
**Pull request by lllHuber**

### CODE REVIEW

**Comment by lllHuber** [comment]
@fraxachun I updated the types of `MenuItem` and `MasterMenuItem` so that I can passt all props to `MenuItemAnchorLink`. Not sure if this creates problems if the two types are merged. What do you think? Merge and pass all props, or keep the types as they are and just pick the props that can be passes to `MenuItemAnchorLink`?

File: packages/admin/cms-admin/src/common/MasterMenu.tsx [hunk 1]
```diff
@@ -37,7 +38,7 @@ export function useMenuFromMasterMenuData(items: MasterMenuData): MenuItem[] {
 }
 
 type MenuItem = {
-    menuItem: MenuItemRouterLinkProps;
+    menuItem: MenuItemRouterLinkProps & MenuItemAnchorLinkProps;
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/common/MasterMenu.tsx:41]: Shouldn't that be `MenuItemRouterLinkProps | MenuItemAnchorLinkProps`?

File: packages/admin/cms-admin/src/common/MasterMenu.tsx [hunk 2]
```diff
@@ -9,7 +10,7 @@ export type MasterMenuItem = Omit<MenuItemRouterLinkProps, "to"> & {
     route?: RouteProps;
     to?: string;
     submenu?: MasterMenuItem[];
-};
+} & MenuItemAnchorLinkProps;
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/common/MasterMenu.tsx:13]: Here as well

**Comment by lllHuber** [comment]
@fraxachun, @johnnyomair could you please have a look at this before the weekend? Would be very helpful 🙏.

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**